test/streaming.test.ts

```diff
@@ -31,16 +31,26 @@ if (isNodeJS()) {
         .topic<Account>('JSforceTestAccountUpdates')
         .subscribe(resolve);
     });
-    await delay(5000);
-    await conn
-      .sobject('Account')
-      .create({ Name: `My New Account #${Date.now()}` });
+    await delay(10000);
+    const accountName = `My New Account #${Date.now()}`;
+
+    await conn.sobject('Account').create({ Name: accountName });
+
     const msg = await msgArrived;
-    assert.ok(isObject(msg.event));
-    assert.ok(msg.event.type === 'created');
-    assert.ok(isObject(msg.sobject));
-    assert.ok(isString(msg.sobject.Name));
-    assert.ok(isString(msg.sobject.Id));
+
+    try {
+      assert.ok(isObject(msg.sobject));
+      assert.ok(msg.sobject.Name === accountName);
+
+      assert.ok(isObject(msg.event));
+      assert.ok(msg.event.type === 'created');
+      assert.ok(isObject(msg.sobject));
+      assert.ok(isString(msg.sobject.Name));
+      assert.ok(isString(msg.sobject.Id));
+    } finally {
+      // cleanup
+      await conn.sobject('Account').findOne({ Name: accountName }).delete();
```

Comment:
this was a bit tricky to make idempotent because the soql query is hardcoded in the `PushTopic` record: https://github.com/jsforce/jsforce/blob/b7ac34f8843ff66670c4473924d663add57e9ea9/scripts/org-setup.mjs#L50-L52

so if the first run failed because the delay wasn't enough, the second run could get the record from the first run:
https://github.com/jsforce/jsforce/actions/runs/6960632515/job/18940569520#step:8:515

this change makes it always delete the record after the test finishes (even in failures) so that the next run only gets one account record.